### PR TITLE
Fix eye icon

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description
Fix visualizations when pressing eye icon. Contains also fix for #1235 

### Checklist
Please include the following checklist in your PR:

- [ ] ~~The `CHANGELOG.md` was updated with the changes introduced in this PR.~~
- [ ] ~~The documentation has been updated if necessary.~~
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has automatic tests where possible.
- [x] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [x] All code has been manually tested in the "debug/interface" scene.
- [ ] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

File: src/rust/ensogl/lib/components/src/toggle_button.rs [hunk 1]
```diff
@@ -202,26 +205,30 @@ impl<Shape:ColorableShape+'static> ToggleButton<Shape>{
 
              // === Input Processing ===
 
+            toggle <- any(...);
```
Comment (MichaelMauderer) [inline src/rust/ensogl/lib/components/src/toggle_button.rs]: I believe having this kind of event is discouraged as it breaks event flow. Instead, there should be a `toggle <- any(...)` that is triggered on any required toggle.

**Review by MichaelMauderer — CHANGES_REQUESTED** [review]

File: src/rust/ensogl/lib/components/src/toggle_button.rs (hunk 1 above)
Comment (wdanilo) [reply src/rust/ensogl/lib/components/src/toggle_button.rs]: I would also go with `toggle <- any(&frp.toggle,&icon.mouse_down);` as it's easier to read the FRP flow then

File: src/rust/ensogl/lib/components/src/toggle_button.rs [hunk 2]
```diff
@@ -202,26 +205,30 @@ impl<Shape:ColorableShape+'static> ToggleButton<Shape>{
...
 
             invisible <- frp.set_visibility.on_false().constant(0.0);
             color.target_alpha <+ invisible;
 
-            visible    <- frp.set_visibility.gate(&frp.set_visibility);
-            is_hovered <- bool(&icon.mouse_out,&icon.mouse_over);
-            is_pressed <- bool(&icon.mouse_up,&icon.mouse_down);
+            frp.source.visible <+ frp.set_visibility;
+            is_hovered         <- bool(&icon.mouse_out,&icon.mouse_over);
+            is_pressed         <- bool(&icon.mouse_up,&icon.mouse_down);
```
Comment (wdanilo) [inline src/rust/ensogl/lib/components/src/toggle_button.rs]: `is_hovered` and `is_pressed` should be exposed from icon directly as this is very common thing to check

**Review by wdanilo — APPROVED** [review]
Looks good to me. I'm not testing it right now though, I will do big testing of the package after all fixes are done.

**Review by MichaelMauderer — APPROVED** [review]

**Pull request merged**